### PR TITLE
Add tBTCv2 pool

### PR DESCRIPTION
**Pull request by lukasz-zimnoch**
### What I did

Added a metapool for [tBTC v2](https://etherscan.io/address/0x18084fbA666a33d37592fA2633fD49a74DD93a88).

### How I did it

The pool is a minimal modification of the existing [tBTC v1 pool](https://github.com/curvefi/curve-contract/tree/master/contracts/pools/tbtc). It uses `sBTC` as a base.

### How to verify it

Confirm that the tests are passing within the CI.

### CODE REVIEW

File: contracts/pools/tbtc2/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+* [`LiquidityGaugeV3`](../../gauges/LiquidityGaugeV3.vy): [TODO](https://etherscan.io/address/TODO)
+* [`StableSwapTBTC2`](StableSwapTBTC2.vy): [TODO](https://etherscan.io/address/TODO)
+
+## Stablecoins
+
+Curve tBTCv2 metapool utilizes the supports swaps between the following assets:
+
+## Direct swaps
+
+Direct swaps are possible between tBTCv2 and the Curve sBTC LP token.
```
Comment (dimpar) [inline contracts/pools/tbtc2/README.md:23]: Just to make sure, "Curve sBTC LP token" is the same as "sbtcCRV", right?

Comment (lukasz-zimnoch) [reply contracts/pools/tbtc2/README.md:23]: Yep

**Comment by dimpar** [comment]
Can we also add a reference to the `tBTCv2` in the main README? https://github.com/keep-network/curve-contract/tree/master/contracts/pools

**Comment by lukasz-zimnoch** [comment]
> Can we also add a reference to the `tBTCv2` in the main README? https://github.com/keep-network/curve-contract/tree/master/contracts/pools

Good point! I'll add it in the PR targeting Curve repo.

File: contracts/pools/tbtc2/DepositTBTC2.vy [hunk 2]
```diff
@@ -0,0 +1,366 @@
+# @version 0.2.7
+"""
+@title "Zap" Depositer for Curve tBTCv2 pool
+@author Curve.Fi
+@license Copyright (c) Curve.Fi, 2020 - all rights reserved
```
Comment (dimpar) [inline contracts/pools/tbtc2/DepositTBTC2.vy]: Not an expert on licensing, but shouldn't we update the year to `2021`?

File: contracts/pools/tbtc2/StableSwapTBTC2.vy [hunk 3]
```diff
@@ -0,0 +1,1078 @@
+# @version 0.2.7
+"""
+@title Curve tBTCv2 Metapool
+@author Curve.Fi
+@license Copyright (c) Curve.Fi, 2020 - all rights reserved
```
Comment (dimpar) [inline contracts/pools/tbtc2/StableSwapTBTC2.vy]: `2021`?

**Comment by dimpar** [comment]
The rest looks good to me!

File: contracts/pools/tbtc2/DepositTBTC2.vy (hunk 2 above)
Comment (lukasz-zimnoch) [reply contracts/pools/tbtc2/DepositTBTC2.vy]: Probably. Will take this into account while opening the PR in Curve repo.

File: contracts/pools/tbtc2/StableSwapTBTC2.vy (hunk 3 above)
Comment (lukasz-zimnoch) [reply contracts/pools/tbtc2/StableSwapTBTC2.vy]: Same as above.

File: .github/workflows/tbtc2.yaml [hunk 4]
```diff
@@ -0,0 +1,59 @@
+name: tbtc2
+
+on:
+  pull_request:
+    paths:
+    - 'tests/**/*.py'
+    - 'contracts/pools/tbtc/**.vy'
```
Comment (tomaszslabon) [inline .github/workflows/tbtc2.yaml]: Shouldn't it be:
```
'contracts/pools/tbtc2/**.vy'
```
?

Comment (lukasz-zimnoch) [reply .github/workflows/tbtc2.yaml]: Good point!

File: .github/workflows/tbtc2.yaml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  push:
+    paths:
+    - 'tests/**/*.py'
+    - 'contracts/pools/tbtc/**.vy'
```
Comment (tomaszslabon) [inline .github/workflows/tbtc2.yaml]: Also here:
tbtc->tbtc2

Comment (lukasz-zimnoch) [reply .github/workflows/tbtc2.yaml]: https://github.com/keep-network/curve-contract/pull/1/commits/eb63230837e5fd8f53c0996b35df70dc267ddb0c

File: contracts/pools/tbtc2/DepositTBTC2.vy (hunk 2 above)
Comment (lukasz-zimnoch) [reply contracts/pools/tbtc2/DepositTBTC2.vy]: https://github.com/keep-network/curve-contract/pull/1/commits/7365c9fa5f28bccd24042e385bb34dc973971a54

File: contracts/pools/tbtc2/StableSwapTBTC2.vy (hunk 3 above)
Comment (lukasz-zimnoch) [reply contracts/pools/tbtc2/StableSwapTBTC2.vy]: https://github.com/keep-network/curve-contract/pull/1/commits/7365c9fa5f28bccd24042e385bb34dc973971a54

**Review by tomaszslabon — APPROVED** [review]

**Comment by lukasz-zimnoch** [comment]
Closed in favor of https://github.com/curvefi/curve-contract/pull/143

**Pull request closed without merging**